pkg/scalers/http_scaler.go

```diff
@@ -0,0 +1,145 @@
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/metrics/pkg/apis/external_metrics"
+	"net/http"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"strconv"
+	"strings"
+)
+
+const (
+	httpMetricName = "HTTPMetricValue"
+)
+
+type httpScaler struct {
+	metadata *httpScalerMetadata
+}
+
+type httpScalerMetadata struct {
+	targetValue int
+	apiURL      string
+	metricName  string
+}
+
+type metric struct {
+	Name  string  `json:"name"`
+	Value float64 `json:"value"`
+}
+
+var httpLog = logf.Log.WithName("http_scaler")
+
+// NewHTTPScaler creates a new HTTP scaler
+func NewHTTPScaler(resolvedEnv, metadata, authParams map[string]string) (Scaler, error) {
+	meta, err := parseHTTPMetadata(resolvedEnv, metadata, authParams)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing HTTP metadata: %s", err)
+	}
+	return &httpScaler{metadata: meta}, nil
+}
+
+func parseHTTPMetadata(resolvedEnv, metadata, authParams map[string]string) (*httpScalerMetadata, error) {
+	meta := httpScalerMetadata{}
+
+	if val, ok := metadata["targetValue"]; ok {
+		targetValue, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("targetValue parsing error %s", err.Error())
+		}
+		meta.targetValue = targetValue
+	} else {
+		return nil, fmt.Errorf("no targetValue given in metadata")
+	}
+
+	if val, ok := metadata["apiURL"]; ok {
+		// remove ending / for better string formatting
+		meta.apiURL = strings.TrimSuffix(val, "/")
+	} else {
+		return nil, fmt.Errorf("no apiURL given in metadata")
+	}
+
+	if val, ok := metadata["metricName"]; ok {
+		meta.metricName = val
+	} else {
+		return nil, fmt.Errorf("no metricName given in metadata")
+	}
+
+	return &meta, nil
+}
+
+func (s *httpScaler) checkHealth() error {
+	u := fmt.Sprintf("%s/health/", s.metadata.apiURL)
+	_, err := http.Get(u)
+	return err
+}
+
+func (s *httpScaler) getMetricInfo() (*metric, error) {
+	var m *metric
+	u := fmt.Sprintf("%s/metrics/%s/", s.metadata.apiURL, s.metadata.metricName)
+	r, err := http.Get(u)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Body.Close()
+	b, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, &m)
+	if err != nil {
+		return nil, err
+	}
+	return m, nil
+}
+
+// Close does nothing in case of httpScaler
+func (s *httpScaler) Close() error {
+	return nil
+}
+
+// IsActive returns true if there are pending messages to be processed
+func (s *httpScaler) IsActive(ctx context.Context) (bool, error) {
+	err := s.checkHealth()
+	if err != nil {
+		httpLog.Error(err, fmt.Sprintf("Error when checking API health: %s", err))
+		return false, err
+	}
+	return true, nil
+}
+
+// GetMetricSpecForScaling returns the MetricSpec for the Horizontal Pod Autoscaler
+func (s *httpScaler) GetMetricSpecForScaling() []v2beta1.MetricSpec {
+	targetQueryValue := resource.NewQuantity(int64(s.metadata.targetValue), resource.DecimalSI)
+	externalMetric := &v2beta1.ExternalMetricSource{
+		MetricName:         httpMetricName,
```

Comment:
`MetricName` should be generated and unique (to allow using multiple Triggers per ScaledObject`). Please refer other scalers in v2, how do the deal with this.

Reply:
@zroubalik do I correctly understand that we are talking about uniqueness on the level of a single type of scaled object per deployment?

Reply:
I think you are talking about different things:
- @zroubalik about the metric name for the trigger in the scaled object
- @turbaszek about the metric name to scrape from the metric source

Is that correct?

Reply:
I think we are talking about the same but I may have not been clear. Will such name be ok?
```go
	metricName := fmt.Sprintf("%s-%s", "http", kedautil.NormalizeString(s.metadata.apiURL), s.metadata.metricName)
	externalMetric := &v2beta2.ExternalMetricSource{
		Metric: v2beta2.MetricIdentifier{
			Name: metricName,
		},
```

Reply:
Yeah, I am talking about the metric name for the trigger in the scaled object (which could be generated from the metric name it will be used to scrape from).

@turbaszek yeah, the metric name should be unique enough, so you could put mutliple triggers of the same type in the scaledobject and you can always distinguish them by the metric name